### DISCUSSION TITLE
Are there some contrib ops that are not yet implemented?

### DISCUSSION THREAD

**Discussion by anirudhsundar** [microsoft/onnxruntime#9270] [Other Q&A] [answered]
Hi all,

I'm new to onnxruntime and am just a user of onnxruntime, so not sure if I'm missing something obvious here. I was trying to use the contrib op `com.microsoft.MulInteger`. I created a `com.microsoft.MulInteger` node in an onnx graph by setting the op type to `MulInteger` and domain to `com.microsoft`. When I try to run this graph with sample inputs, I'm seeing the below error

```
Current thread 0x00007fe6ad7ab740 (most recent call first):
  File "/home/temp/anaconda3/lib/python3.7/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 321 in _create_inference_session
  File "/home/temp/anaconda3/lib/python3.7/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 283 in __init__
  File "/home/temp/anaconda3/lib/python3.7/site-packages/onnxruntime/backend/backend.py", line 107 in prepare
  File "test_ort.py", line 137 in get_onnxruntime_output
  File "test_ort.py", line 200 in test_mulinteger
```

Grepping through the source shows only a couple of files where the word `MulInteger` is used. One looks like the op definition schema in [quantization_defs.cc](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/graph/contrib_ops/quantization_defs.cc#L280) and the other is the documentation itself in [ContribOperators.md](https://github.com/microsoft/onnxruntime/blob/master/docs/ContribOperators.md#com.microsoft.MulInteger).

- Could someone explain how contrib operators are added?
- Is it that a new op definition is initially created and documented, and then later the implementation of the op is added?

Thanks in advance.

**Comment by yufenglee** [comment, marked as answer]
ReduceSumInteger and MulInteger were designed but not implemented. Initially we experiment those ops and want to integrate them to ONNX as standard ops. We now tends to use the Quantize/Dequantize (fakequant in other framework) format to represent quantized model.    We will remove those ops schema.

**Reply by anirudhsundar** [reply]
Thanks for the reply @yufenglee
